File: TASK08/opcode/opcode.cpp
#pragma once
#include <iostream>
using namespace std;

typedef int Opcode;

const Opcode InputByte = 1;    // [code] [offset]
const Opcode OutputByte = 2;    // [code] [offset]
const Opcode ChangeValue = 3;  // [code] [offset] [amount]
const Opcode SetValue = 4;     // [code] [offset] [amount]
const Opcode JumpIfZero = 5;   // [code] [offset] [jump-offset]
const Opcode JumpIfNotZero = 6; // [code] [offset] [jump-offset]

File: TASK07/targets/opcode/opcode.go
// Package opcode implements the opcode compilation target. It provides
// structures and methods to compile an instruction.Chunk into opcode and
// to run the same.
package opcode

// Opcode represents a single opcode instruction.
type Opcode int

// Various opcode instructions.
const (
	_ Opcode = iota

	InputByte     // [code] [offset]
	OutputByte    // [code] [offset]
	ChangeValue   // [code] [offset] [amount]
	SetValue      // [code] [offset] [amount]
	JumpIfZero    // [code] [offset] [jump-offset]
	JumpIfNotZero // [code] [offset] [jump-offset]
)
